File: app/src/main/java/com/stupidtree/hichat/utils/JsonUtils.java
package com.stupidtree.hichat.utils;

import com.google.gson.JsonObject;

/**
 * 此类整合了一些JSON格式数据解析有关的函数
 */
public class JsonUtils {
    public static Integer getIntegerData(JsonObject jo,String key){
        try {
            return jo.get(key).getAsInt();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }


    public static String getStringData(JsonObject jo,String key){
        try {
            return jo.get(key).getAsString();
        } catch (Exception ignored) {
            return null;
        }
    }

    public static JsonObject getObjectData(JsonObject jo,String key){
        try {
            return jo.get(key).getAsJsonObject();
        } catch (Exception e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/com/stupidtree/cloudliter/utils/JsonUtils.kt
package com.stupidtree.cloudliter.utils

import com.google.gson.JsonObject
import org.json.JSONException
import org.json.JSONObject

/**
 * 此类整合了一些JSON格式数据解析有关的函数
 */
object JsonUtils {
    fun getIntegerData(jo: JsonObject?, key: String?): Int? {
        return try {
            jo?.get(key)?.asInt?:0
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }

    fun getStringData(jo: JsonObject?, key: String?): String? {
        return try {
            jo?.get(key)?.asString
        } catch (ignored: Exception) {
            null
        }
    }

    fun getObjectData(jo: JsonObject?, key: String?): JsonObject? {
        return try {
            jo?.get(key)?.asJsonObject
        } catch (e: Exception) {
            e.printStackTrace()
            null
        }
    }


    fun getJSONObject(str: String?): JSONObject? {
        return try {
            str?.let {
                JSONObject(it)
            }
        } catch (e: JSONException) {
            null
        }
    }
}